include/m_string.h

```diff
@@ -342,7 +342,7 @@ static inline void human_readable_num_bytes(char *buf, int buf_len,
   for (i = 0; dbl_val > 1024 && i < sizeof(size) - 1; i++) dbl_val /= 1024;
   const char mult = size[i];
   // 18446744073709551615 Yottabytes should be enough for most ...
-  if (dbl_val > ULLONG_MAX)
+  if (dbl_val > static_cast<double>(ULLONG_MAX))
```

Comment:
Although this might not be important in this particular comparison, this transformation is not quite accurate.
ULLONG_MAX (64-bit, all ones) cannot be represented precisely in 64-bit floating point (with 52 bit mantissa).
May be consider `dbl_val >= ldexp(1, sizeof(ulonglong) * 8)`

Reply:
You are right that `ULLONG_MAX` is different from `static_cast<double>(ULLONG_MAX)`.
I fixed `if (dbl_val > ULLONG_MAX)` to `if (dbl_val >= ULLONG_MAX)`, but other places seems to be fine.
It looks that `ldexp` doesn't give us any advantages over `static_cast<double>`:
```
ULLONG_MAX=18446744073709551615
static_cast<double>(ULLONG_MAX)=18446744073709551616.000000
ldexp(1, sizeof(ulonglong)*8)=18446744073709551616.000000
LLONG_MAX=9223372036854775807
static_cast<double>(LLONG_MAX)=9223372036854775808.000000
ldexp(1, sizeof(ulonglong)*8 - 1)=9223372036854775808.000000
LLONG_MIN=-9223372036854775808
static_cast<double>(LLONG_MIN)=-9223372036854775808.000000
-ldexp(1, sizeof(ulonglong)*8 - 1)=-9223372036854775808.000000
```